src/context/StateProvider.js

```diff
@@ -9,24 +17,33 @@ const StateProvider = ({ children }) => {
   const initialProductState = {
     cart: [],
     wishlist: [],
-    productList: [],
+    productsList: [],
   };
 
-  const [{ cart, wishlist, productList }, productDispatch] = useReducer(
-    productReducer,
+  const [{ cart, wishlist, productsList }, productsDispatch] = useReducer(
+    productsReducer,
     initialProductState
   );
 
+  useEffect(() => {
+    axios.get("/api/products").then((response) => {
+      productsDispatch({
+        type: "SET_PRODUCTS",
+        payload: response.data.products,
+      });
+    });
+  }, []);
```

Comment:
Suggestion:- It is good to use async-await for API calls.

Reply:
Yup, already on it. Thank you!